vnext/ReactWindowsCore/WinRTWebSocketResource.cpp

```diff
@@ -66,14 +66,16 @@ namespace
 
       void await_suspend(std::experimental::coroutine_handle<> resume)
       {
-        m_queue.Post([context = resume.address()]() noexcept
+        m_callBack = [context = resume.address()]() noexcept
```

Comment:
>m_callBack [](start = 8, length = 10)

This doesn't look right. Why don't we want to allocate on stack?

Reply:
Because it seems that we have a race condition here. See:[https://devblogs.microsoft.com/oldnewthing/20191223-00/?p=103255](https://devblogs.microsoft.com/oldnewthing/20191223-00/?p=103255)

The relevant part is:
This code violates one of the rules we gave when we were getting started with awaitable objects: Once you arrange for the handle to be called, you cannot access any member variables because the coroutine may have resumed before async_suspend finishes.

Our theory for the crash when we currently call the Mso::VoidFunctor destructor on the call stack is that the destructor is called after the coroutine handle is called from another thread.
So, what we are trying to do here is to avoid the Mso::VoidFunctor destructor call after we called m_queue.Post. Instead we put the temporary Mso::VoidFunctor to a field. In that case the callstack is safe to be destroyed any time after we call the m_queue.Post.

---
In reply to: [420211611](https://github.com/microsoft/react-native-windows/pull/4790#discussion_r420211611) [](ancestors = 420211611)

Reply:
Ok, looks like you thought about it in much greater detail. Perhaps would be nice to go over such pattern on one of deep dives :)

---
In reply to: [420376594](https://github.com/microsoft/react-native-windows/pull/4790#discussion_r420376594) [](ancestors = 420376594,420211611)

Reply:
@JunielKatarn, consider to add a comment to the code explaining why we use field instead of a temporary variable.

Reply:
@vmoroz Already merged. Will add the comment on the next PR.